bgpd/bgp_zebra.c

```diff
@@ -750,6 +750,14 @@ bool bgp_zebra_nexthop_set(union sockunion *local, union sockunion *remote,
 								? peer->conf_if
 								: peer->ifname,
 							peer->bgp->vrf_id);
+			else if (peer->update_if)
+				ifp = if_lookup_by_name(peer->update_if,
+							peer->bgp->vrf_id);
+			else
+				ifp = if_lookup_by_ipv6_exact(
```

Comment:
I'm ok with getting the correct interface from the `peer->update_if` as that is associated with the `update-source` cli but given the way LL works in v6 world doing a if_lookup_by_ipv6_exact will more than likely not return the correct interface here and will lead to weird issues and this else statement should be removed.

Reply:
I think that would be great to be able to configure neighbor with hardset LL, but by mentioning the interface.
for instance:
```
neighbor fe80::4113:5bba:2b61:b20c%eth0 remote-as 55
```

Reply:
ok. removed call to ipv6_exact, and kept the check with update-source command.

Reply:
@donaldsharp , please review